src/equations.jl

```diff
@@ -126,6 +126,10 @@ for T in [:Num, :Complex, :Number], S in [:Num, :Complex, :Number]
     end
 end
 
+canonical_form(eq::Equation) = eq.lhs - eq.rhs ~ 0
+
+get_variables(eq::Equation) = unique(vcat(get_variables(eq.lhs), get_variables(eq.rhs)))
```

Comment:
@ValentinKaisermayer, @YingboMa, @shashi Just note this is a breaking change as previously `get_variables` only applied to the rhs. It also means `get_variables` is no longer consistent with `get_variables!`. Please be aware of this in making any new Symbolics releases. (It may not have any practical effect -- so far I haven't found anything that really relied on that behavior but I could have missed something in ModelingToolkit or Catalyst...)

Reply:
I guess there has been a release already and this was included, so it's a bit late to make sure the release is flagged as breaking...

Reply:
How can it be breaking if `get_variables(eq::Equation)` is added and not altered?

Reply:
It was previously (intentionally) defined with a different behavior:

https://github.com/JuliaSymbolics/Symbolics.jl/blob/0026df9dfa4a97dec7f578c1cae28f58445f2f71/src/utils.jl#L74

https://github.com/JuliaSymbolics/Symbolics.jl/blob/0026df9dfa4a97dec7f578c1cae28f58445f2f71/src/utils.jl#L78

The new dispatch changes this behavior.

Reply:
I see. We can remove it, it is after all just for convenience and not used in MTK. Only `get_variables!`: https://github.com/SciML/ModelingToolkit.jl/blob/28b36c8f9932acfc6c54737c51437d35b192b0d7/src/systems/dependency_graphs.jl#L43
However, I must say, that without proper documentation it is not clear why only the rhs is considered. I guess it assumes `0~f(x)` or `D(x)~f(x)`, but just by convention. 
https://symbolics.juliasymbolics.org/stable/manual/expression_manipulation/#Symbolics.get_variables

Reply:
I think the change makes sense given ModelingToolkit is now allowing more flexibility in equation structure, but the in-place version should probably get updated for consistency. I agree the behavior should get documented in either case though!